File: Examples/Simple Integration Obj-C/Source/Models/OrderAmount/OrderAmount.m
//
//  OrderAmount.m
//  Simple Integration Obj-C
//
//  Created by Johnny Peter on 29/08/19.
//  Copyright © 2019 Network International. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "OrderAmount.h"

@implementation OrderAmount

@synthesize currencyCode;
@synthesize value;


-(instancetype)initWitCurrencyCode:(NSString *)currencyCode andValue:(int) value {
    if(self = [super init]) {
        self.currencyCode = currencyCode;
        self.value = value;
    }
    return self;
}


-(NSDictionary *)dictionary {
    return [[NSDictionary alloc] initWithObjectsAndKeys: self.currencyCode, @"currencyCode", [NSNumber numberWithInt: self.value],  @"value", nil];
}

@end

File: Examples/Simple Integration/Source/Models/OrderAmount.swift
//
//  OrderAmount.swift
//  Simple Integration
//
//  Created by Johnny Peter on 23/08/19.
//  Copyright © 2019 Network International. All rights reserved.
//

import Foundation

struct OrderAmount: Encodable {
    let currencyCode: String?
    let value: Double?
    
    private enum AmountCodingKeys: String, CodingKey {
        case currencyCode
        case value
    }
}
